In [3]:
import numpy as np
import pandas as pd

In [4]:
df = pd.read_csv("loan_data_set.csv")

In [5]:
df.columns

Index(['Loan_ID', 'Gender', 'Married', 'Dependents', 'Education',
       'Self_Employed', 'ApplicantIncome', 'CoapplicantIncome', 'LoanAmount',
       'Loan_Amount_Term', 'Credit_History', 'Property_Area', 'Loan_Status'],
      dtype='object')

In [6]:
Y = df['Loan_Status']
X = df.drop(columns =['Loan_ID','Loan_Status'])
print(X.columns)

Index(['Gender', 'Married', 'Dependents', 'Education', 'Self_Employed',
       'ApplicantIncome', 'CoapplicantIncome', 'LoanAmount',
       'Loan_Amount_Term', 'Credit_History', 'Property_Area'],
      dtype='object')


In [7]:
#X.describe()
#df['Gender'].unique()

In [8]:
total=0
count=0
indices=[]
for index,rows in df.iterrows():
    if(rows['Gender']!='Male' and rows['Gender']!='Female'):
        count+=1
        indices.append(index)
    total+=1
print(count/total)
print(indices)

0.021172638436482084
[23, 126, 171, 188, 314, 334, 460, 467, 477, 507, 576, 588, 592]


In [9]:
df['Married'].unique()

array(['No', 'Yes', nan], dtype=object)

In [10]:
total=0
count=0
indices=[]
for index,rows in df.iterrows():
    if(rows['Married']!='Yes' and rows['Married']!='No'):
        count+=1
        indices.append(index)
    total+=1
print(count/total)
print(indices)

0.004885993485342019
[104, 228, 435]


In [11]:
df['Dependents'].unique()

array(['0', '1', '2', '3+', nan], dtype=object)

In [12]:
total=0
count=0
indices=[]
for index,rows in df.iterrows():
    if(rows['Dependents'] not in ['0','1','2','3+']):
        count+=1
        indices.append(index)
    total+=1
print(count/total)
print(indices)

0.024429967426710098
[102, 104, 120, 226, 228, 293, 301, 332, 335, 346, 355, 435, 517, 571, 597]


In [13]:
df['Education'].unique()

array(['Graduate', 'Not Graduate'], dtype=object)

In [14]:
df['Self_Employed'].unique()

array(['No', 'Yes', nan], dtype=object)

In [15]:
total=0
count=0
indices=[]
for index,rows in df.iterrows():
    if(rows['Self_Employed'] not in ['Yes','No']):
        count+=1
        indices.append(index)
    total+=1
print(count/total)
print(indices)

0.05211726384364821
[11, 19, 24, 29, 30, 95, 107, 111, 114, 158, 170, 218, 231, 236, 268, 295, 333, 336, 344, 374, 380, 385, 411, 432, 447, 463, 468, 535, 542, 579, 600, 601]


In [16]:
import math
total=0
count=0
indices=[]
for index,rows in df.iterrows():
    if(math.isnan(rows['ApplicantIncome'])):
        count+=1
        indices.append(index)
    total+=1
print(count/total)
print(indices)

0.0
[]


In [17]:
import math
total=0
count=0
indices=[]
for index,rows in df.iterrows():
    if(math.isnan(rows['CoapplicantIncome'])):
        count+=1
        indices.append(index)
    total+=1
print(count/total)
print(indices)

0.0
[]


In [18]:
total=0
count=0
indices=[]
for index,rows in df.iterrows():
    if(math.isnan(rows['LoanAmount'])):
        count+=1
        indices.append(index)
    total+=1
print(count/total)
print(indices)

0.035830618892508145
[0, 35, 63, 81, 95, 102, 103, 113, 127, 202, 284, 305, 322, 338, 387, 435, 437, 479, 524, 550, 551, 605]


In [19]:
total=0
count=0
indices=[]
for index,rows in df.iterrows():
    if(math.isnan(rows['Loan_Amount_Term'])):
        count+=1
        indices.append(index)
    total+=1
print(count/total)
print(indices)

0.02280130293159609
[19, 36, 44, 45, 73, 112, 165, 197, 223, 232, 335, 367, 421, 423]


In [20]:
total=0
count=0
indices=[]
for index,rows in df.iterrows():
    if(math.isnan(rows['Credit_History'])):
        count+=1
        indices.append(index)
    total+=1
print(count/total)
print(indices)

0.08143322475570032
[16, 24, 30, 42, 79, 83, 86, 95, 117, 125, 129, 130, 156, 181, 187, 198, 219, 236, 237, 259, 260, 279, 309, 313, 317, 318, 323, 348, 363, 377, 392, 395, 411, 444, 449, 451, 460, 473, 490, 491, 497, 503, 506, 530, 533, 544, 556, 565, 583, 600]


In [21]:
df['Property_Area'].unique()

array(['Urban', 'Rural', 'Semiurban'], dtype=object)

In [22]:
total=0
count=0
indices=[]
for index,rows in df.iterrows():
    if(rows['Property_Area'] not in ['Urban','Rural','Semiurban']):
        count+=1
        indices.append(index)
    total+=1
print(count/total)
print(indices)

0.0
[]


### Exploratory Data Analysis
<table>
    <tr>
        <th>Column</th>
        <th>Unique values</th>
        <th>% of NaN</th>
        <th>Indices of NaN</th>
    </tr>
    <tr>
        <td>Gender</td>
        <td>Male,Female</td>
        <td>0.021</td>
        <td>[23, 126, 171, 188, 314, 334, 460, 467, 477, 507, 576, 588, 592]</td>
    </tr>
    <tr>
        <td>Married</td>
        <td>Yes,No</td>
        <td>0.0048</td>
        <td>[104, 228, 435]</td>
    </tr>
    <tr>
        <td>Dependents</td>
        <td>0,1,2,3+</td>
        <td>0.024</td>
        <td>[102, 104, 120, 226, 228, 293, 301, 332, 335, 346, 355, 435, 517, 571, 597]</td>
    </tr>
    <tr>
        <td>Education</td>
        <td>Graduate, Not Graduate</td>
        <td>0.0</td>
        <td>[]</td>
    </tr>
    <tr>
        <td>Self_Employed</td>
        <td>Yes, No</td>
        <td>0.052</td>
        <td>[11, 19, 24, 29, 30, 95, 107, 111, 114, 158, 170, 218, 231, 236, 268, 295, 333, 336, 344, 374, 380, 385, 411, 432, 447, 463, 468, 535, 542, 579, 600, 601]</td>
    </tr>
    <tr>
        <td>Applicant_Income</td>
        <td>some range</td>
        <td>0.0</td>
        <td>[]</td>
    </tr>
    <tr>
        <td>Coapplicant_Income</td>
        <td>some range</td>
        <td>0.0</td>
        <td>[]</td>
    </tr>
    <tr>
        <td>Loan_Amount</td>
        <td>some range</td>
        <td>0.036</td>
        <td>[0, 35, 63, 81, 95, 102, 103, 113, 127, 202, 284, 305, 322, 338, 387, 435, 437, 479, 524, 550, 551, 605]</td>
    </tr>
    <tr>
        <td>Loan_Amount_Term</td>
        <td>some range</td>
        <td>0.023</td>
        <td>[19, 36, 44, 45, 73, 112, 165, 197, 223, 232, 335, 367, 421, 423]</td>
    </tr>
    <tr>
        <td>Credit_History</td>
        <td>some range</td>
        <td>0.081</td>
        <td>[16, 24, 30, 42, 79, 83, 86, 95, 117, 125, 129, 130, 156, 181, 187, 198, 219, 236, 237, 259, 260, 279, 309, 313, 317, 318, 323, 348, 363, 377, 392, 395, 411, 444, 449, 451, 460, 473, 490, 491, 497, 503, 506, 530, 533, 544, 556, 565, 583, 600]</td>
    </tr>
    <tr>
        <td>Property_Area</td>
        <td>Urban,Rural,SemiUrban</td>
        <td>0.0</td>
        <td>[]</td>
    </tr>
</table>

In [23]:
df['Loan_Amount_Term'].unique()

array([360., 120., 240.,  nan, 180.,  60., 300., 480.,  36.,  84.,  12.])